### ISSUE TITLE
[Issue]: Asterisk in filename causes unnecessary .NFO file update

### ISSUE THREAD

**Issue by scott-ainsworth** [jellyfin/jellyfin#10887]
### Problem Summary

I have a curious situation that causes one particular NFO to be recreated although nothing about the episode has changed. Here are some things I believe are relevant:

- The TV episode file is named `Doc Martin (2004) - S01E03 - Sh*t happens!.mkv`. As is normal, the file is stored in the `Season 01` subdirectory.
- JellyFin is set to automatically create and update NFO files.
- Finally, I create a file called `sha256sums.txt` in each directory. This file contains checksums for every media file in the subdirectory.

### Reproduction

When the `sha256sums.txt` is updated, a refresh event is created as shown in the log:

`[2024-01-18 16:01:05.412 -05:00] [DBG] [4] Emby.Server.Implementations.IO.LibraryMonitor: New file refresher created for "/Volumes/Media/Library/Shows/Doc Martin (2004)/Season 01/sha256sums.txt"`

Shortly thereafter, a refresh is executed:

`[2024-01-18 16:02:55.880 -05:00] [INF] [34] Emby.Server.Implementations.IO.LibraryMonitor: "Season 1" ("/Volumes/Media/Library/Shows/Doc Martin (2004)/Season 01") will be refreshed.`

The strange part is that the episode NFO file is updated even though the only change is the addition or update of the `sha256sums.txt` file.

Of course, this creates a race condition: updates to the NFO cause updates to `sha256sums.txt`, which cause udpates to the NFO, and so on.

I suspect the root cause is that the filename is used unescaped, which causes the asterisk(*) to be treated as a wild card.

Thanks,
Scott

### Jellyfin Version

Other

### if other:

10.8.11

### Environment

```markdown
- OS: macOS Sonoma 14.2.1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felix920506** [comment]
Is there any reason for not using the latest Jellyfin release (10.8.13)?

**Comment by solidsnake1298** [comment]
Jellyfin's docs say to avoid special characters, and specifically calls out asteriks.

https://jellyfin.org/docs/general/server/media/shows/

![image](https://github.com/jellyfin/jellyfin/assets/137357145/3e11ffc1-2986-471f-8513-93e29b39e908)

**Comment by scott-ainsworth** [comment]
@felix920506: Thanks. I hadn't realized a point release was available. (Of course, it happens with 10.8.13.)

**Comment by scott-ainsworth** [comment]
@solidsnake1298: I read the docs so long ago that had forgotten about the "avoid" note. Even so, this behavior is unexpected. There are several reasons I prefer special characters:

- Some software insists on using the filenames as titles, even when the metadata contains a title. This is almost universally true for MKV files.
- I have script that sets the title from filenames. In past I tried substituting non-problematic characters for problematic characters (e.g., "_" for ":") and letting the script translate. Unfortunately, I ran out non-used, non-problematic characters to serve as substitutes.

Except for this issue, special characters have not been a problem. So, as a work around, I'll probably move the checksum files to a parallel directory tree.

Finally, is the file and path handling code centralized or scattered about the code.  If it is centralized, I will consider taking a look.

Thanks,
Scott G. Ainsworth
